towersOfHanoi/main.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+// assert library is used for testing
+const assert = require('assert');
+const readline = require('readline');
+const { start } = require('repl');
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout
+});
+
+let stacks = {
+  a: [4, 3, 2, 1],
+  b: [],
+  c: []
+};
+
+
+const printStacks = () => {
+  console.log("a: " + stacks.a);
+  console.log("b: " + stacks.b);
+  console.log("c: " + stacks.c);
+}
+
+let n = 1;
+
+const movePiece = (startStack, endStack) => stacks[endStack].push(stacks[startStack].pop())
+  // let element = stacks[startStack].pop();
+    
+
+const isLegal = (startStack, endStack) => {
+  let element1 = [stacks[startStack].length-1];
```

Comment:
this check is not enough, since it will allow you to move a bigger piece on top of a smaller peice